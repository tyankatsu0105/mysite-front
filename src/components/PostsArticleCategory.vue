<template>
	<div class="PostsArticleCategory">
		<FontAwesomeIcon
			:icon="{ prefix: 'fa', iconName: 'folder' }"
		/>
		<p class="PostsArticleCategory-CategoryText">
			{{ text }}
		</p>
	</div>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faFolder } from "@fortawesome/free-solid-svg-icons";

export default {
  components: {
    FontAwesomeIcon
  },
  props: {
    text: {
      type: String,
      requirer: true,
      default: ""
    }
  },
  created() {
    library.add(faFolder);
  }
};
</script>

<style scoped lang='scss'>
.PostsArticleCategory {
  &-CategoryText {
    display: inline-block;
    margin-left: 5px;
    font-size: 1rem;
  }
}
</style>
