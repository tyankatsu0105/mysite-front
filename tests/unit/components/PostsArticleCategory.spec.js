import { shallowMount } from "@vue/test-utils";
import PostsArticleCategory from "@/components/PostsArticleCategory.vue";

const propsData = {
  text: "カテゴリ"
};

describe("PostsArticleCategory", () => {
  it("render", () => {
    const wrapper = shallowMount(PostsArticleCategory, {
      propsData,
      stubs: ["font-awesome-icon"]
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
