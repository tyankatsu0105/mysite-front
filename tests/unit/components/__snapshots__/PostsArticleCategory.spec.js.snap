// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsArticleCategory render 1`] = `
<div class="PostsArticleCategory">
  <fontawesomeicon-stub icon="[object Object]"></fontawesomeicon-stub>
  <p class="PostsArticleCategory-CategoryText">
    カテゴリ
  </p>
</div>
`;
